Adjust dartdevc documentation for tool deprecation (#4003)
dart-lang/site-www: src/_guides/language/type-system.md

## Runtime checks

> Runtime checks deals with type safety issues

Reviewer: ```suggestion
Runtime checks deal with type safety issues
```

that can't be detected at compile time.

For example, the following code throws an exception at runtime

> because it is an error to cast a list of dogs to a list of cats:

Reviewer: ```suggestion
because it's an error to cast a list of dogs to a list of cats:
```
